protocol/0082-RFPR-on_chain_referral_program.md

```diff
@@ -0,0 +1,238 @@
+# On Chain Referral Program Specification
+
+## Summary
+
+The aim of the on-chain referral program is to allow users of the protocol to incentivise users and community members to refer new traders by voting to provide benefits for referrers and/or referees. 
+
+Whilst a referral program is active, the following benefits may be available to eligible parties:
+
+- a **referrer** will receive a proportion of all referee taker fees as a **reward**.
+- a **referee** will be eligible for a **discount** on any taker fees they incur.
+
+The size of the reward or discount will be dependent on the benefit tier the referee's [team](#team-mechanics) is currently placed in. A team becomes eligible for greater benefits by reaching a minimum running volume over a number of epochs. Both referrers and referees contribute to their team's running volume.
+
+On-chain referral programs can be created and updated through [governance proposals](#governance-proposals).
+
+## Network Parameters
+
+- `referralProgram.maxBenefitTiers` - limits the maximum number of [benefit tiers](#benefit-tiers) which can be specified as part of a referral program
+- `referralProgram.maxReferralRewardFactor` - limits the maximum reward factor which can be specified as part of a referral program
+- `referralProgram.maxReferralDiscountFactor` - limits the maximum discount factor which can be specified as part of a referral program governance proposal
+- `referralProgram.maxPartyVolumePerEpoch` - limits the volume in quantum units which is eligible each epoch for referral program mechanisms
+- `referralProgram.minStakedVegaTokens` - limits team creation to parties staking at least this number of tokens
```

Comment:
@barnabee / @cdummett / @TomMcL do you see any value to have, on top of this minimum, a requirement for extra vega tokens for each team member on the team?

Reply:
Can definitely see the value, adds a disadvantage to teams merging to access higher tiers which I think is good

I think though if we add this we will need to add supporting features for the referrer like being able to remove "dead-weight" team members and possibly "approve" new team members which is added complexity for what I feel should be a fairly simple referral program (at least at first).

Reply:
@barnabee says - not needed.